spec/resource/application_spec.rb

```diff
@@ -524,21 +523,32 @@
   end
 
   describe '#verification_emails' do
-    let(:directory_with_verification) { test_directory_with_verification }
+    let(:directory_with_verification) do
+      test_api_client.directories.create(name: random_directory_name('ruby'),
+                                         description: 'email verification enabled')
+    end
 
     before do
-      test_api_client.account_store_mappings.create({ application: app, account_store: directory_with_verification,
-        list_index: 1, is_default_account_store: false, is_default_group_store: false })
+      test_api_client.account_store_mappings.create(application: app,
+                                                    account_store: directory_with_verification,
+                                                    list_index: 1,
+                                                    is_default_account_store: false,
+                                                    is_default_group_store: false)
+
+      directory_with_verification.account_creation_policy.verification_email_status = 'ENABLED'
+      directory_with_verification.account_creation_policy.verification_success_email_status = 'ENABLED'
+      directory_with_verification.account_creation_policy.welcome_email_status = 'ENABLED'
+      directory_with_verification.account_creation_policy.save
```

Comment:
this should be moved as a method to a class in support/ folder. the same code is used above and will be probably used in other places